ksql-engine/src/main/java/io/confluent/ksql/planner/plan/KsqlStructuredDataOutputNode.java

```diff
@@ -114,46 +114,36 @@ public KeyField getKeyField() {
     final QueryContext.Stacker contextStacker = builder.buildNodeContext(getId());
 
     final Set<Integer> rowkeyIndexes = SchemaUtil.getRowTimeRowKeyIndexes(getSchema());
-    final Builder outputNodeBuilder = new Builder(this);
     final Schema schema = SchemaUtil.removeImplicitRowTimeRowKeyFromSchema(getSchema());
-    outputNodeBuilder.withSchema(schema);
 
     final SchemaKStream<?> result = createOutputStream(
         schemaKStream,
-        outputNodeBuilder,
         builder.getKsqlConfig(),
         builder.getFunctionRegistry(),
         contextStacker
     );
 
-    final KsqlStructuredDataOutputNode noRowKey = outputNodeBuilder.build();
-    final KsqlTopicSerDe ksqlTopicSerDe = noRowKey
-        .getKsqlTopic()
+    final KsqlTopicSerDe ksqlTopicSerDe = getKsqlTopic()
         .getKsqlTopicSerDe();
 
     final Serde<GenericRow> outputRowSerde = builder.buildGenericRowSerde(
         ksqlTopicSerDe,
-        noRowKey.getSchema(),
+        schema,
         contextStacker.getQueryContext());
 
     result.into(
-        noRowKey.getKafkaTopicName(),
+        getKafkaTopicName(),
         outputRowSerde,
         rowkeyIndexes
     );
 
-    result.setOutputNode(
-        outputNodeBuilder
-            .withSchema(
-                SchemaUtil.addImplicitRowTimeRowKeyToSchema(noRowKey.getSchema()))
```

Comment:
This change of schema can be removed as the `addImplicitRowTimeRowKeyToSchema` call is now done in the `PhysicalPlanBuilder` when building the `PersistentQueryMetadata`.